src/data/finsupp/basic.lean

```diff
@@ -1565,14 +1565,50 @@ finset.subset.trans support_sum $
   finset.subset.trans (finset.bUnion_mono $ assume a ha, support_single_subset) $
   by rw [finset.bUnion_singleton]; exact subset.refl _
 
-lemma map_domain_support_of_injective [decidable_eq β] {f : α → β} (hf : function.injective f)
-  (s : α →₀ M) : (map_domain f s).support = finset.image f s.support :=
+lemma map_domain_apply' {α β M : Type*} [add_comm_monoid M] (S : set α) {f : α → β} (x : α →₀ M)
+  (hS : (x.support : set α) ⊆ S) (hf : set.inj_on f S) {a : α} (ha : a ∈ S) :
+  map_domain f x (f a) = x a :=
+begin
+  rw [map_domain, sum_apply, sum],
+  simp_rw single_apply,
+  have : ∀ (b : α) (ha1 : b ∈ x.support),
+    (if f b = f a then x b else 0) = if f b = f a then x a else 0,
```

Comment:
```suggestion
  have hfa : ∀ (b ∈ x.support), (if f b = f a then x b else 0) = if f b = f a then x a else 0,
```
(new name because otherwise later on we have two variables named `this`)

Reply:
Is `this` overloading such a big problem? It always seemed like quite common practice to me, and a good way of indicating that a subproof is temporary and won't be used later, (which is indeed the case for both `this`'s here)

Reply:
Fair question. I wouldn't say it's a "big problem" but I think it is better to avoid ever having more than one variable in the context with the same name.

I agree this is subjective, I can believe it is common practice, and I see no advice about it in our [style guide](https://leanprover-community.github.io/contribute/style.html) so you're certainly within your rights to stick to your naming, but IMHO it is better to avoid such overloading.

If I really wanted to convey the hint as you suggest, I think I'd `clear this` after the first use.

Reply:
I'll leave it as is, maybe for other named have statement's I'd want to avoid overloading, but there are certainly plenty of examples of `have := blah, have := foo this, have := quux this` type-chains in mathlib so I hope this isn't too irregular.